beacon-chain/sync/initial-sync/round_robin.go

```diff
@@ -189,6 +139,29 @@ func (s *Service) processFetchedData(
 	}
 }
 
+// processFetchedData processes data received from queue.
+func (s *Service) processFetchedDataRegSync(
+	ctx context.Context, genesis time.Time, startSlot uint64, data *blocksQueueFetchedData) {
+	defer func() {
+		if !featureconfig.Get().EnablePeerScorer || data.pid == "" {
+			return
+		}
+		scorer := s.p2p.Peers().Scorers().BlockProviderScorer()
+		if diff := s.chain.HeadSlot() - startSlot; diff > 0 {
```

Comment:
Could this underflow? Maybe change to "if diff is not zero" 

```suggestion
		if diff := s.chain.HeadSlot() - startSlot; diff != 0 {
```

Reply:
Actually, what about `s.chain.HeadSlot() != startSlot`?

```suggestion
		if s.chain.HeadSlot() != startSlot {
```

Reply:
Will fix both in a moment.